mne/io/eeglab/tests/test_eeglab.py

```diff
@@ -55,6 +55,7 @@ def _check_h5(fname):
 
 @requires_h5py
 @testing.requires_testing_data
+@pytest.mark.filterwarnings('ignore:Function read_events_eeglab is deprecated')
```

Comment:
`pytest.deprecated_call` is probably better as it's more targeted and shows us what we can delete later more clearly

Reply:
And if you nest them they not always pass 'cos is catching stuff:
```sh

――――――――――――――――――――――――――――――――― test_io_set_raw[fnames0] ―――――――――――――――――――――――――――――――――

fnames = ['/home/sik/mne_data/MNE-testing-data/EEGLAB/test_raw.set', '/home/sik/mne_data/MNE-testing-data/EEGLAB/test_raw_onefile.set']
tmpdir = '/tmp/pytest-of-sik/pytest-4/test_io_set_raw_fnames0_0'

    @requires_h5py
    @testing.requires_testing_data
    @pytest.mark.parametrize('fnames', [raw_mat_fnames, raw_h5_fnames])
    def test_io_set_raw(fnames, tmpdir):
        """Test importing EEGLAB .set files."""
        tmpdir = str(tmpdir)
        raw_fname, raw_fname_onefile = fnames
        with pytest.warns(RuntimeWarning) as w:
            _test_raw_reader(read_raw_eeglab, input_fname=raw_fname,
                             montage=montage)
            _test_raw_reader(read_raw_eeglab, input_fname=raw_fname_onefile,
                             montage=montage)
        for want in ('Events like', 'consist entirely', 'could not be mapped',
                     'string preload is not supported'):
            assert (any(want in str(ww.message) for ww in w))
    
        with pytest.warns(RuntimeWarning) as w:
            with pytest.deprecated_call():
                # test finding events in continuous data
                event_id = {'rt': 1, 'square': 2}
                raw0 = read_raw_eeglab(input_fname=raw_fname, montage=montage,
                                       event_id=event_id, preload=True)
                raw1 = read_raw_eeglab(input_fname=raw_fname, montage=montage,
                                       event_id=event_id, preload=False)
                raw2 = read_raw_eeglab(input_fname=raw_fname_onefile,
                                       montage=montage, event_id=event_id)
                raw3 = read_raw_eeglab(input_fname=raw_fname, montage=montage,
                                       event_id=event_id)
                raw4 = read_raw_eeglab(input_fname=raw_fname, montage=montage)
    
            assert raw0.filenames[0].endswith('.fdt')  # .set with additional .fdt
            assert raw2.filenames[0].endswith('.set')  # standalone .set
    
            Epochs(raw0, find_events(raw0), event_id)
            epochs = Epochs(raw1, find_events(raw1), event_id)
    
            assert len(find_events(raw4)) == 0  # no events without event_id
            assert epochs["square"].average().nave == 80   # 80 with
            assert_array_equal(raw0[:][0], raw1[:][0], raw2[:][0], raw3[:][0])
>           assert_array_equal(raw0[:][-1], raw1[:][-1], raw2[:][-1], raw3[:][-1])
E           Failed: DID NOT WARN. No warnings of type (<class 'RuntimeWarning'>,) was emitted. The list of emitted warnings is: [].

epochs     = <Epochs  |   154 events (good & bad), -0.203125 - 0.5 sec, baseline [None, 0], ~85 kB, data not loaded,
 'rt': 74
 'square': 80>
event_id   = {'rt': 1, 'square': 2}
fnames     = ['/home/sik/mne_data/MNE-testing-data/EEGLAB/test_raw.set', '/home/sik/mne_data/MNE-testing-data/EEGLAB/test_raw_onefile.set']
raw0       = <RawEEGLAB  |  test_raw.fdt, n_channels x n_times : 33 x 30504 (238.3 sec), ~7.8 MB, data loaded>
raw1       = <RawEEGLAB  |  test_raw.fdt, n_channels x n_times : 33 x 30504 (238.3 sec), ~85 kB, data not loaded>
raw2       = <RawEEGLAB  |  test_raw_onefile.set, n_channels x n_times : 33 x 30504 (238.3 sec), ~7.8 MB, data loaded>
raw3       = <RawEEGLAB  |  test_raw.fdt, n_channels x n_times : 33 x 30504 (238.3 sec), ~85 kB, data not loaded>
raw4       = <RawEEGLAB  |  test_raw.fdt, n_channels x n_times : 33 x 30504 (238.3 sec), ~85 kB, data not loaded>
raw_fname  = '/home/sik/mne_data/MNE-testing-data/EEGLAB/test_raw.set'
raw_fname_onefile = '/home/sik/mne_data/MNE-testing-data/EEGLAB/test_raw_onefile.set'
tmpdir     = '/tmp/pytest-of-sik/pytest-4/test_io_set_raw_fnames0_0'
w          = WarningsChecker(record=True)
want       = 'string preload is not supported'

mne/io/eeglab/tests/test_eeglab.py:96: Failed

```

I would let it as it is, it works for now. And I would take a further look. Unless, someone has strong oposition to it.

Reply:
here's the ticket https://github.com/mne-tools/mne-python/pull/5532